src/main/java/dev/andrylat/vomelianchuk/CardNumberOperations/enums/PaymentSystem.java

```diff
@@ -0,0 +1,34 @@
+package dev.andrylat.vomelianchuk.CardNumberOperations.enums;
+
+import java.util.Optional;
+
+public enum PaymentSystem {
+    VISA(4),
+    MASTERCARD(51, 52, 53, 54, 55),
+    DINERS_CLUB(36, 38),
+    DISCOVER(6011, 65),
+    JCB(35),
+    AMERICAN_EXPRESS(34, 37);
+
+    private final int[] prefixes;
+
+    PaymentSystem(int... prefixes) {
+        this.prefixes = prefixes;
+    }
+
+    public int[] getPrefixes() {
+        return prefixes;
+    }
+
+    public static Optional<PaymentSystem> detectPaymentSystem(String cardNumber) {
+        if (cardNumber == null) return Optional.empty();
```

Comment:
we discussed last time that you should always use {} for each operator like if, while, for, etc.